pandas/core/apply.py

```diff
@@ -541,17 +545,26 @@ def maybe_apply_str(self) -> Optional[FrameOrSeriesUnion]:
         f = self.f
         if not isinstance(f, str):
             return None
+
+        obj = self.obj
+
+        # TODO: GH 39993 - Avoid special-casing by replacing with lambda
+        if f == "size" and isinstance(obj, ABCDataFrame):
+            # Special-cased because DataFrame.size returns a single scalar
+            value = obj.shape[self.axis]
+            return obj._constructor_sliced(value, index=self.agg_axis, name="size")
```

Comment:
Is it intentional to set the name here? This isn't consistent with other string agg methods

Reply:
Indeed, thanks @TheNeuralBit! Opened #42046

Reply:
Thank you!